docs/middleware.rst

```diff
@@ -0,0 +1,113 @@
+Middlewares
+###########
+
+Middlewares are an essential part of the event sending lifecycle as it's processed
+through them before being sent to the server. Each middleware wraps the next to
+resemble an onion structure. There are several built-in middlewares whose list
+is:
+
+- ``BreadcrumbInterfaceMiddleware``: it adds all the recorded breadcrumbs up to
+  the point the event was generated to the event.
+- ``ContextInterfaceMiddleware``: it adds the data stored in the contexts to the
+  event.
+- ``ExceptionInterfaceMiddleware``: it fetches the stacktrace for the captured
+  exception (if any) and integrates additional data like a small piece of source
+  code present around each stackframe.
+- ``MessageInterfaceMiddleware``: it adds a message (if present) to the event
+  and optionally format it using ``vsprintf``.
+- ``ModulesMiddleware``: it fetches informations about the packages installed
+  through Composer. The ``composer.lock`` file must be present to make this
+  middleware working.
+- ``ProcessorMiddleware``: it's a special middleware that executes the registered
+  processors by passing to them the event instance.
```

Comment:
Why is it "special" it executes registered processors, is that special?

Reply:
No it's not, I've rephrased the whole sentence and it should be better now